app/controllers/application_controller.rb

```diff
@@ -21,6 +21,10 @@ def parse_json_request
   end
 
   def encoded_base_path
-    URI.escape(params[:base_path])
+    URI.escape(base_path)
+  end
+
+  def base_path
+    "/#{params[:base_path]}"
```

Comment:
I do wonder whether this param name is still right. It feels that it could be confusing that `params[:base_path]` doesn't actually contain a base_path.  Maybe we should rename it to something like `base_path_without_slash` to make it very clear.